contrib/fwupd.spec.in

```diff
@@ -334,6 +334,7 @@ done
 %files -f %{name}.lang
 %doc README.md AUTHORS
 %license COPYING
+%{_sysconfdir}/ld.so.conf.d/fwupd.conf
```

Comment:
Hmm, I was hoping that we could do it without dropping a file in `ld.so.conf.d`.  Can we not do a dynamic load of the built in library from our fixed location like we do with the other plugins?

Reply:
Or perhaps can we use `rpath`?

Reply:
> Or perhaps can we use rpath?

Nah, that's not going to work for flatpak and also gets stripped out for Fedora/RHEL. What's the problem with the ld.so.conf.d out of interest? I guess we could dynamically load all symbols but I'm not sure what advantage it gives us over just using the runtime linker.

Reply:
> Nah, that's not going to work for flatpak and also gets stripped out for Fedora/RHEL. 

But for flatpak you could wrap an environment variable to set the `LD_LIBRARY_PATH` instead.  What strips it for Fedora/RHEL?  It seems like a nice way to keep a private search path.

> What's the problem with the ld.so.conf.d out of interest?

I was thinking we can avoid polluting the space of all the applications that look for libraries from the runtime linker.

I did try this on top of this branch and confirmed colorhug loads from fwupdtool:

```
diff --git a/src/meson.build b/src/meson.build
index 53b470c74..0ba5c9449 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -194,18 +194,6 @@ fwupdpluginbuiltin = library(
   ],
 )

-# we're installing the fwupdpluginbuiltin to the plugin_dir to avoid causing Debian ABI issues,
-# so tell ld where it can be found
-con2 = configuration_data()
-con2.set('FWUPD_PLUGINDIR', plugin_dir)
-configure_file(
-  input: 'fwupd.conf',
-  output: 'fwupd.conf',
-  configuration: con2,
-  install: true,
-  install_dir: join_paths(sysconfdir, 'ld.so.conf.d')
-)
-
 fwupdtool = executable(
   'fwupdtool',
   resources_src,
@@ -238,6 +226,7 @@ fwupdtool = executable(
     fwupdpluginbuiltin,
   ],
   install: true,
+  install_rpath : plugin_dir,
   install_dir: bindir
 )
```

Reply:
At least looking at https://fedoraproject.org/wiki/Changes/Broken_RPATH_will_fail_rpmbuild it seems to me rpath should be allowable in RPMs as long as it's pointing somewhere safe, no?

Reply:
Maybe it's okay then? https://fedoraproject.org/wiki/RPath_Packaging_Draft seems super vague -- but I think https://docs.fedoraproject.org/en-US/packaging-guidelines/ allows it for this kind of usecase?

Reply:
Can you push that change to my branch as a fixup and I'll try a test Fedora package to see what kind of abuse I get from the build system. It might be okay!

Reply:
The case of `Rpath for Internal Libraries` seems to match exactly what we're discussing IMO.

Reply:
If we want to make libfwupdplugin _internal_ it seems this is probably the way to go about it too.  
_Realistically_ if we're making all of this change, I have to question: should we just be linking it to libfwupdplugin instead and our "ABI" is only for the symbols not linked to it by these static libraries.

Reply:
> Can you push that change to my branch as a fixup and I'll try a test Fedora package to see what kind of abuse I get from the build system. It might be okay!

done

Reply:
If that works, I just pushed to your branch what I think is the next logical improvement based on it.